app/src/main/java/nyc/c4q/jonathancolon/inContaq/utlities/graphs/WeeklyGraph.java

```diff
@@ -0,0 +1,240 @@
+    private static final int WED = 4;
+    private static final int THUR = 5;
+    private static final int FRI = 6;
+    private static final int SAT = 7;
+    private static final int DEFAULT_VALUE = 0;
+    private final String TAG = "sms";
+    private Context context;
+    private int highestValue;
+    private LineChartView lineGraph;
+    private float[] receivedValues;
+    private float[] sentValues;
+    private ArrayList<Sms> lstSms;
+
+    public WeeklyGraph(Context context, LineChartView lineGraph, ArrayList<Sms> lstSms) {
+
+        this.context = context;
+        this.lineGraph = lineGraph;
+        this.lstSms = lstSms;
+    }
+
+    public void showWeeklyGraph() {
+
+        getLineGraphValues(lstSms);
+        getHigehstValueForYaxis();
+        loadGraph();
+    }
+
+    private void getLineGraphValues(ArrayList<Sms> lstSms) {
+
+        receivedValues = setValues(getDailyReceived(lstSms));
+        sentValues = setValues(getMonthlySent(lstSms));
+    }
+
+    private float[] setValues(TreeMap<Integer, Integer> numberOfTexts) {
+
+        ArrayList<Float> list = new ArrayList<Float>();
+        for (Map.Entry<Integer, Integer> entry : numberOfTexts.entrySet()) {
+
+            Float value = entry.getValue().floatValue();
+            list.add(value);
+        }
+        return convertFloats(list);
+    }
+
+    private TreeMap<Integer, Integer> getDailyReceived(ArrayList<Sms> texts) {
+
+        TreeMap<Integer, Integer> monthlyReceived = setUpDailyTextMap();
+        MonthlyTaskParams receivedParams = new MonthlyTaskParams(texts, monthlyReceived);
+        MonthlyReceivedWorkerTask monthlyReceivedTask = new MonthlyReceivedWorkerTask();
+
+        try {
+
+            monthlyReceived = monthlyReceivedTask.execute(receivedParams).get();
+            Log.e(TAG, MONTHLY_RECEIVED + monthlyReceived);
+
+        } catch (InterruptedException | ExecutionException e) {
+
+            e.printStackTrace();
+        }
+        return monthlyReceived;
+    }
+
+    private TreeMap<Integer, Integer> getMonthlySent(ArrayList<Sms> texts) {
+
+        TreeMap<Integer, Integer> monthlySent = setUpDailyTextMap();
+        MonthlyTaskParams sentParams = new MonthlyTaskParams(texts, monthlySent);
+        MonthlySentWorkerTask monthlySentWorkerTask = new MonthlySentWorkerTask();
+
+        try {
+
+            monthlySent = monthlySentWorkerTask.execute(sentParams).get();
+            Log.e(TAG, MONTHLY_SENT + monthlySent);
+
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        }
+        return monthlySent;
+    }
+
+    // This converts the generic "Float" to primitive float because the graphs only uses primitives
+    private float[] convertFloats(List<Float> floats) {
+
+        float[] ret = new float[floats.size()];
+        Iterator<Float> iterator = floats.iterator();
+        for (int i = 0; i < ret.length; i++) {
+            ret[i] = iterator.next().intValue();
```

Comment:
Do we need an iterator and iterator.next() if we're just looping from 0 to the length of ret, which is same length as floats?